### PR TITLE
ref(breakdowns): All fields in breakdown config should be camelCase, and rename the breakdown key name in project options.

### PR DESCRIPTION
**Pull request by dashed**
Replacement for https://github.com/getsentry/relay/pull/993 with applied suggestions:

- Hard rename the breakdown key in project state options from `breakdown` to `breakdown_v2` (Sentry will provide the config using its camel-case name `breakdownV2`).

- I've added the serde attribute to catch for all unknown enum variants for `enum BreakdownConfig`.

### CODE REVIEW

File: relay-general/src/store/normalize/breakdowns.rs [hunk 1]
```diff
@@ -202,15 +202,19 @@ impl EmitBreakdowns for SpanOperationsConfig {
 
 /// Configuration to define breakdown to be generated based on properties and breakdown type.
 #[derive(Debug, Clone, Serialize, Deserialize)]
-#[serde(tag = "type", rename_all = "snake_case")]
+#[serde(tag = "type", rename_all = "camelCase")]
 pub enum BreakdownConfig {
```
Comment (dashed) [inline relay-general/src/store/normalize/breakdowns.rs:206]: @untitaker I couldn't really use `enum ErrorBoundary<T>` in the project config (https://github.com/getsentry/relay/blob/a231a2b17a71304a19710cf9a49b21fa06e25f71/relay-general/src/store/mod.rs#L55) as it's in a different crate (https://github.com/getsentry/relay/blob/a231a2b17a71304a19710cf9a49b21fa06e25f71/relay-server/src/utils/error_boundary.rs).

I'm assuming this is what you had in mind. 

If so, we might need to consolidate `enum ErrorBoundary<T>` somewhere, and implement `Default` for `enum ErrorBoundary<T>`.

File: relay-general/src/store/normalize/breakdowns.rs [hunk 2]
```diff
@@ -202,15 +202,19 @@ impl EmitBreakdowns for SpanOperationsConfig {
...
     SpanOperations(SpanOperationsConfig),
+    #[serde(other)]
+    Unsupported,
 }
 
 impl EmitBreakdowns for BreakdownConfig {
     fn emit_breakdowns(&self, event: &Event) -> Option<Measurements> {
         match self {
             BreakdownConfig::SpanOperations(config) => config.emit_breakdowns(event),
+            _ => None,
```
Comment (untitaker) [inline relay-general/src/store/normalize/breakdowns.rs]:
```suggestion
            BreakdownConfig::Unsupported => None,
```

If new breakdown variant is added, one should not forget to implement it..

**Review by untitaker — APPROVED** [review]

File: relay-general/benches/benchmarks.rs [hunk 3]
```diff
@@ -90,7 +90,7 @@ fn bench_store_processor(c: &mut Criterion) {
         user_agent: None,
         sent_at: None,
         received_at: None,
-        breakdowns: None,
+        breakdowns_v2: None,
```
Comment (jan-auer) [inline relay-general/benches/benchmarks.rs]: The store config does not have to be renamed, since this is just an internal type.

**Review by jan-auer — CHANGES_REQUESTED** [review]
Looks good. The "V2" looks a bit strange at first, but is a good and explicit solution.

One change requested below, since this PR touches more places than required.

Note that once you merge and deploy this, Relay will no longer compute breakdowns until the corresponding Sentry PR is merged.

**Comment by dashed** [comment]
>  @jan-auer  Note that once you merge and deploy this, Relay will no longer compute breakdowns until the corresponding Sentry PR is merged.

Yep, I'm aware. That should be fine as the computed breakdown are not critical to the product.

File: relay-general/benches/benchmarks.rs (hunk 3 above)
Comment (dashed) [reply relay-general/benches/benchmarks.rs]: You're right. I've renamed this back to `breakdowns`.

**Review by jan-auer — APPROVED** [review]
Thanks!

@untitaker can you please also confirm that you're +1 on keeping StoreConfig as-is?

**Review by untitaker — APPROVED** [review]

**Comment by untitaker** [comment]
I presume the places where we still have this as public API don't get breakdowns anyway

**Pull request merged**